src/products/products.controller.ts

```diff
@@ -36,7 +46,7 @@ export class ProductsController {
   async getProducts(): Promise<ProductDto[]> {
     this.logger.debug('Get all products.');
     const allProducts = await this.productsService.findAll();
-    return allProducts.map<ProductDto>(ProductDto.covertToApi);
+    return allProducts.map<ProductDto>(this.convertToApi);
```

Comment:
Don't pass instance methods as arguments, it might lead to unexpected behavior.
```suggestion
    return allProducts.map((product: Product) => this.convertToApi(product));
```

Reply:
@derevnjuk can't we just do `return allProducts.map<ProductDto>((p: Product) => new ProductDto(p));` and not use an extra function at all?

Reply:
Sure, that's what I suggest you above  😄 
 https://github.com/NeuraLegion/brokencrystals/pull/62#discussion_r679214745